### DISCUSSION TITLE
Tailwind pseudo elements classes - range thumb

### DISCUSSION THREAD

**Discussion by zenicek** [tailwindlabs/tailwindcss#10508] [Help] [answered]
Hello,

I have been reading quite a few discussions, yet I am to find a workable solution for customizing the thumb on the input range slider. I must be missing something here.

Here is the simple situation. I have an input slider which user can customize to an extent, they are able to select an accent color. Because tailwind does not support runtime variables and I don't know what color user selects from the the palette, I am using the inline style to set the accent, but also our design has the custom size thumb. So what I am trying to achieve is basically having custom width and height of the thumb.

Here is a playground.

https://play.tailwindcss.com/uaLlgivInA

Now I have read through [this thread](https://github.com/tailwindlabs/tailwindcss/discussions/6972) and it does indeed work when passing all of the pseudo classes to the element, it makes it nice and pink, but I cannot use it in my case as thumb needs to be of the accent color (same as filled in bar of the user option).

I have also attempted to do something like this in global.css:

`  .thumb-lg::-webkit-slider-thumb {
    @apply h-7 w-7;
  }`

Of course it did not work and the class is not generated. I have also attempted to do the same through plugin, without any result, or [using variants](https://tailwindcss.com/docs/hover-focus-and-other-states#using-arbitrary-variants). However, if I remove the ::-webkit-slider-thumb the class is being generated just fine, so the setup is working.

Appreciate any help.

**Comment by wongjn** [comment, marked as answer]
You need to apply `appearance-none` to the `::-webkit-slider-thumb` element. However, as you discovered, `accent-color` would no longer have an effect. Instead, you could apply a `background-color` to the `::-webkit-slider-thumb` via a custom CSS property which then allows you to dynamically set the value of the custom CSS property in `style`. Here's [a Tailwind Play](https://play.tailwindcss.com/vnvE2pL6kE) of the technique.

**Reply by zenicek** [reply]
That is awesome, worked very well, thank you 😄
